### ISSUE TITLE
KeyError: 'bias' while running show_weights(clf, vec=vec) in xgboost notebook

### ISSUE THREAD

**Issue by SaiAparna1** [TeamHG-Memex/eli5#372]
Could anyone please help me to solve following error,

  File "C:\anaconda3\lib\site-packages\xgboost\sklearn.py", line 745, in intercept_
    return np.array(json.loads(b.get_dump(dump_format='json')[0])['bias'])

KeyError: 'bias'

**Comment by SusmithBarigidad** [comment]
KeyError : 'bias' is shown when we run the XGBRegressor, please help us on this error

**Comment by sugarcrm-aorso** [comment]
Seeing the same issue on an XGBClassifier().

**Comment by danielad-k** [comment]
I have the same issue as well with XGBClassifier.

**Comment by AshwinB-hat** [comment]
![image](https://user-images.githubusercontent.com/33116373/80304078-92db9480-87d1-11ea-9ce5-e72549ad760a.png)
I just tested the XgbRegressor and could not reproduce the issue. 
Can one of you paste a screenshot of what you is happening?
@SusmithBarigidad  @sugarcrm-aorso  @danielad-k

**Comment by supinyu** [comment]
Python Version
Python 3.6.4 |Anaconda, Inc.| (default, Jan 16 2018, 18:10:19) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> exit()

![image](https://user-images.githubusercontent.com/12369067/80369653-3657a280-88c1-11ea-80f3-e3a523352d5d.png)

**Comment by sugarcrm-aorso** [comment]
> ![image](https://user-images.githubusercontent.com/33116373/80304078-92db9480-87d1-11ea-9ce5-e72549ad760a.png)
> I just tested the XgbRegressor and could not reproduce the issue.
> Can one of you paste a screenshot of what you is happening?
> @SusmithBarigidad @sugarcrm-aorso @danielad-k

It looks like it may be related to the Classifier while you are referencing the Regressor. Let me know if you need anything more beyond what @supinyu shared.

**Comment by JulienB-78** [comment]
The bug is not on eli5 but on xgboost. The call `hasattre(xgb, 'intercept_')` crashes. This bug is present in the 1.0.2 release and has been fixed by commit [Fix checking booster. (#5505)](https://github.com/dmlc/xgboost/commit/dc2950fd90bd8d73f2c6131679179094b49e7289). It will be integrated in the 1.1.0 version in preparation (1st release candidate already out).

In the mean time you can easily apply the patch yourself (just one line to change in `intercept_(self)` of site-packages/xgboost/sklearn.py). Replace
     `getattr(self, 'booster', None) is not None and self.booster not in {gbtree', 'dart'}:`
by
     `self.get_params()['booster'] not in {'gbtree', 'dart'}:`.

**Comment by supinyu** [comment]
Thank you very much JulienB-78.
I update the xgboost version to 1.1.0rc1. The code will not have this error.

**Comment by sugarcrm-aorso** [comment]
Yup, using the 1.1.0 release candidate worked for me as well. Thanks @JulienB-78

**Comment by Naqi444** [comment]
> The bug is not on eli5 but on xgboost. The call `hasattre(xgb, 'intercept_')` crashes. This bug is present in the 1.0.2 release and has been fixed by commit [Fix checking booster. (#5505)](https://github.com/dmlc/xgboost/commit/dc2950fd90bd8d73f2c6131679179094b49e7289). It will be integrated in the 1.1.0 version in preparation (1st release candidate already out).
> 
> In the mean time you can easily apply the patch yourself (just one line to change in `intercept_(self)` of site-packages/xgboost/sklearn.py). Replace
> `getattr(self, 'booster', None) is not None and self.booster not in {gbtree', 'dart'}:`
> by
> `self.get_params()['booster'] not in {'gbtree', 'dart'}:`.

Does this update work for anyone? Does not seem to be working for me. 
I could upgrade my xgboost to 1.1.0 but I have another dependency which has a requirement of 1.0.0
